design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+## Test plan
+
+Before we dig into which tests should run, we should decide where, and how, they
+should run. Currently, the `conjur-authn-k8s-client` have UTs in its project
+and a vanilla test that runs in `kubernetes-conjur-demo`. The `kubernetes-conjur-demo`
+is triggered only when we push code to that repository. 
+
+Even while putting aside the fact that we have only a vanilla test, our current
+ test flow is still not optimal. 
+ Let's propose some options to improve our flow, and decide on the best one.
+
+### Trigger `kubernetes-conjur-demo` to run after every build of `conjur-authn-k8s-client`
+
+We can find a way to run `kubernetes-conjur-demo` using the currently built
+ authenticator-client, utilizing the `LOCAL_AUTHENTICATOR` environment variable.
+ If this is possible, we can trigger a `kubernetes-conjur-demo` build as part
+  of our build.
+
+This will give us full confidence in the green build as it also passed
+ integration tests.
+ 
+ However, the build time of `kubernetes-conjur-demo` is ~25 minutes (opposing
+  2-3 minutes of the `conjur-authn-k8s-client`) and it is not very stable
+  . This means that it will harder to merge PRs into the `conjur-authn-k8s
+  -client`.
+  
+I am not sure that this can be implemented. By setting`LOCAL_AUTHENTICATOR
+` to `true` we use the last built image (ran in the `./bin/build` step. In
+ case we have builds running in parallel it may affect the latest built image
+  so it is possible that we can't use a local authenticator in Jenkins.
+
+### Trigger `kubernetes-conjur-demo` to run after every master build of `conjur-authn-k8s-client`
+
+This option will still not fail builds that will fail in `kubernetes-conjur-demo`
+but we will get _some_ feedback as at the end of the day we will see which PR
+ introduced a failure and it will be easier to fix the error. 
+  
+### Add integration tests to `conjur-authn-k8s-client` as we have in the `secrets-provider-for-kubernetes`
+  
+In the `secrets-provider-for-kubernetes` we have integration tests
+with different scenarios, that each one deploys its own pod and verifies
+the expected output.
+
+The caveat of this approach is that members of the community will not be able
+to contribute to this repo as they can today as they will not be able to run
+the integration tests (as they run with `summon`). They will not be able to
+fix tests that broke because of their chane and will not be able to add tests
```

Comment:
"because of their chane"? 

toolchain?

Reply:
change :)